### PR TITLE
maint: Use ansys actions in workflow

### LINKED ISSUE #9: Update workflow actions inline with rest of API packages
Based on internal conversation. The workflows for the API packages can be homogenized by using a common action that would take care of the build process. Let's get that action running.

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
Changes the workflow file of this repo to use the ansys actions. Closes #9 .

### CODE REVIEW

**Review by RobPasMue — COMMENTED** [review]
@AlejandroFernandezLuces - base yourself on https://github.com/ansys/ansys-templates/blob/feat/template-api/src/ansys/templates/python/ansys_api/%7B%7B%20cookiecutter.__project_name_slug%20%7D%7D/.github/workflows/ci.yml

I recommend you to follow this workflow and replicate it. This is going to be the new standard

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,123 +2,71 @@ name: Ansys Prime Server gRPC API CI
 
 on:
   pull_request:
-    branches:
-      - main
-
   push:
     tags:
-      - "v*"
+      - "*"
```
Comment (ninad-kamat) [inline .github/workflows/ci.yml:7]: Keep this as `v*`. We only want to run when a release tag is pushed.

**Review by ninad-kamat — DISMISSED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ninad-kamat) [reply .github/workflows/ci.yml:7]: This is a protected tag in policies as well.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,123 +2,71 @@ name: Ansys Prime Server gRPC API CI
...
-          twine upload --skip-existing ./**/*.tar.gz
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.TWINE_TOKEN }}
+      - name: "Release to the private PyPI"
+        uses: ansys/actions/release-pypi-private@v4
+        with:
+          library-name: ${{ env.LIBRARY_NAME }}
+          twine-username: "__token__"
+          twine-token: ${{ secrets.PYANSYS_PYPI_PRIVATE_PAT }}
```
Comment (RobPasMue) [inline .github/workflows/ci.yml:64]:
```suggestion
      - name: "Release to the public PyPI"
        uses: ansys/actions/release-pypi-public@v4
        with:
          library-name: ${{ env.LIBRARY_NAME }}
          twine-username: "__token__"
          twine-token: ${{ secrets.TWINE_TOKEN }}
```

Comment (RobPasMue) [reply .github/workflows/ci.yml:64]: It has to be released to the public PyPI

**Review by RobPasMue — CHANGES_REQUESTED** [review]

**Review by RobPasMue — APPROVED** [review]

**Pull request merged**